File: android/core/src/main/java/com/americangroupllc/buddyplay/core/observability/CrashReportingService.kt
package com.americangroupllc.buddyplay.core.observability

/**
 * Crash + non-fatal-error reporting slot. v1 attaches
 * [NoopCrashReportingService]; v1.1 will optionally attach Sentry behind a
 * Settings opt-in.
 *
 * IMPORTANT: BuddyPlay v1 does NOT send any crash reports.
 */
interface CrashReportingService {
    fun capture(error: Throwable, context: Map<String, String> = emptyMap())
    fun breadcrumb(message: String, category: String)
}

class NoopCrashReportingService : CrashReportingService {
    override fun capture(error: Throwable, context: Map<String, String>) {}
    override fun breadcrumb(message: String, category: String) {}
}

File: apps/BuddyPlay/shared/BuddyCore/Sources/BuddyCore/Observability/CrashReportingService.swift
import Foundation

/// Crash + non-fatal-error reporting slot. v1 attaches
/// `NoopCrashReportingService`; v1.1 will optionally attach Sentry behind a
/// Settings opt-in.
///
/// IMPORTANT: BuddyPlay v1 does NOT send any crash reports. This file
/// declares the interface only.
public protocol CrashReportingService {
    func capture(error: Error, context: [String: String])
    func breadcrumb(_ message: String, category: String)
}

public final class NoopCrashReportingService: CrashReportingService {
    public init() {}
    public func capture(error: Error, context: [String: String]) {}
    public func breadcrumb(_ message: String, category: String) {}
}

#if canImport(Sentry)
// Real implementation lands in v1.1 once the user opts in via Settings.
// import Sentry
// public final class SentryCrashReportingService: CrashReportingService { ... }
#endif
